### PR TITLE
[Github actions] Run test suite on pull requests

### PR DESCRIPTION
**Pull request by thecristen**
**Asana ticket:** [Run test suite on pull requests](https://app.asana.com/0/555089885850811/1200285402859674)

The `tests.yml` workflow ("CI") has been enabled for a few weeks now -- what do you think?

This PR has some documentation that goes with these tests, along with a bunch of cleanup. It's _probably_ easier to review if you examine the changes one commit at a time.

* Does the documentation make sense / is it helpful?
* Does each test script make sense and work as expected?
* Did I accidentally remove anything you depend on?
* Would anything else be helpful here?

### Side note

PRs to `master` will now require that all required checks are passing before allowing a merge.
"all required checks" consists of the "Build docker image" workflow and every job in "CI" _except_ the Wallaby tests (since I haven't fixed those yet). Skipped jobs are considered passing.

### CODE REVIEW

**Review by alicia-mbta — APPROVED** [review]
As someone who has never used GitHub actions, this looks intuitive enough for me to understand, more or less. I'm curious about how you were making sure that everything was working as expected as you were making these changes?
In any case, it all looks good to me.
_Semaphore is (almost) dead, long live GitHub actions!_

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -381,38 +298,26 @@ jobs:
...
       with:
         path: "apps/site/*/node_modules"
         key: ${{ runner.os }}-nodejs-${{ hashFiles('apps/site/*/package-lock.json') }}
-    - run: npx prettier "{js,ts}/**/*.{js,ts,tsx}" --list-different
-      working-directory: apps/site/assets/
+    - run: npm run ci:format:ts
 
   elixir_wallaby:
-    name: Integration tests / Elixir
+    name: Integration tests / Elixir (FIXME)
```
[the remark this answers is not in the document]
Comment (thecristen) [reply .github/workflows/tests.yml:309]: sadly yes :P

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -216,16 +174,12 @@ jobs:
...
           key: ${{ runner.os }}-nodejs-${{ hashFiles('apps/site/*/package-lock.json') }}
       - name: Build app
-        run: |
-          mix phx.digest
-          npm run react:build
-          npm run webpack:build
-          mix compile --warnings-as-errors
+        run: npm run build
       - env:
           RECAPTCHA_PUBLIC_KEY: 6LeIxAcTAAAAAJcZVRqyHh71UMIEGNQ_MXjiZKhI
```
[the remark this answers is not in the document]
Comment (thecristen) [reply .github/workflows/tests.yml:179]: Yeah the Recaptcha ones are super public

File: docs/TESTING.md [hunk 3]
```diff
@@ -1,83 +1,133 @@
-# Tests (WIP)
+# Tests
```
[the remark this answers is not in the document]
Comment (thecristen) [reply docs/TESTING.md:1]: > I'm curious about how you were making sure that everything was working as expected as you were making these changes?

Too much trial and error. And to be honest I wouldn't be surprised if I find a new more tiny things to change after more PRs get to try this workflow. :)

File: README.md [hunk 4]
```diff
@@ -129,25 +129,17 @@ any given time, the site may not be compatible with the very latest API version
...
-    npm run react:setup && npm run react:build
-    ```
-
-1.  Build the assets:
+1.  Build the project:
     ```
-    npm run webpack:build
+    npm run build
     ```
+  This does several things: builds the Phoenix application assets, builds all the front-end assets, and then compiles the entire Elixir application.
```
[the remark this answers is not in the document]
Comment (thecristen) [reply README.md:142]: gotta love commands that run more commands

**Review by hannahpurcell — APPROVED** [review]
Woo!  Testing.md is awesome to read.  Learned so much, and explains why tests are skipped (hadn't made that connection yet).  Thanks!

**Pull request merged**